### ISSUE TITLE
Panel processes clicks on disabled buttons etc

### ISSUE THREAD

**Issue by xnsde** [holoviz/panel#4301]
#### ALL software version info
Panel 0.14.2

#### Description of expected behavior and the observed behavior
If I disable a button in the python code, I would expect it to be non-clickable.
It is, but this is done purely by setting the HTML attribute `disabled="true"` 
If you inspect in the browser, then delete the `disabled="true"`, the button becomes clickable and panel processes events.

I would expect panel to know it's disabled and refuse to process any events from disabled items - alternative document very carefully that the user needs to check for disabled themselves everywhere they use it.

#### Complete, minimal, self-contained example code that reproduces the issue

```python
import panel as pn

txt = pn.widgets.TextInput()
btn = pn.widgets.Button(name="Disabled button")
btn.disabled = True


def callback(event):
    txt.value = f"Button clicked: {event}"


btn.on_click(callback)

pn.Row(btn, txt).servable()
```

Steps:
1) Run and open app (panel serve $filename)
2) Right-Click button, inspect
3) Find <button> tag and remove the `disabled="true"` part.
4) Click button and observe text in text panel changes

#### Screenshots or screencasts of the bug in action
![image](https://user-images.githubusercontent.com/111856701/213384324-f138b30b-daca-4e23-a3a2-8f849d1fcef6.png)
![image](https://user-images.githubusercontent.com/111856701/213384387-6c4b37ca-759c-4fac-8202-b9c7604f7a86.png)

**Comment by jbednar** [comment]
I'm not sure it's reasonable for Panel to be defending against people live-editing its code; seems like there's no way to stop that in general.

**Comment by MarcSkovMadsen** [comment]
I was also thinking about this. 

Would it be possible on serverside 
to not accept or just skip events from client if a widget is disabled?

**Comment by philippjfr** [comment]
As a general matter we will never guard against all such cases, but I would certainly be quite happy to accept a PR that disables events triggered from disabled widgets. The only concern is that we cannot guard against this at the Bokeh level which means that we will continue to sync properties of disabled widgets (not an issue for a Button which does not have state). This means that in theory you can end up in a situation where the state of the widget becomes desynchronized from anything attached to it, I wouldn't rate that very highly though.

TL;DR I would be happy to accept a PR that disables processing of events attached to a disabled widget.
